assets/js/payment-methods-demo/express-payment/apple-pay/apple-pay-express.js

```diff
@@ -0,0 +1,380 @@
+	activePaymentMethod,
+	setActivePaymentMethod,
+} ) => {
+	/**
+	 * @type {[ StripePaymentRequest|null, function( StripePaymentRequest ):StripePaymentRequest|null]}
+	 */
+	// @ts-ignore
+	const [ paymentRequest, setPaymentRequest ] = useState( null );
+	const stripe = useStripe();
+	const [ canMakePayment, setCanMakePayment ] = useState( false );
+	const eventHandlers = useRef( DEFAULT_STRIPE_EVENT_HANDLERS );
+	const currentBilling = useRef( billing );
+	const currentShipping = useRef( shippingData );
+	const currentPaymentRequest = useRef( paymentRequest );
+	const currentPaymentStatus = useRef( paymentStatus );
+	const currentEventRegistration = useRef( eventRegistration );
+	const isActive = useRef( activePaymentMethod === PAYMENT_METHOD_NAME );
+
+	// update refs when any change
+	useEffect( () => {
+		currentBilling.current = billing;
+		currentShipping.current = shippingData;
+		currentPaymentRequest.current = paymentRequest;
+		currentPaymentStatus.current = paymentStatus;
+		currentEventRegistration.current = eventRegistration;
+		isActive.current = activePaymentMethod === PAYMENT_METHOD_NAME;
+	}, [ billing, shippingData, paymentRequest, paymentStatus, isActive ] );
+
+	//set paymentRequest
+	useEffect( () => {
+		// can't do anything if stripe isn't available yet or we have zero total
+		if ( ! stripe || ! billing.cartTotal.value ) {
+			return;
+		}
+
+		// if payment request hasn't been set yet then set it
+		if ( ! currentPaymentRequest.current ) {
+			setPaymentRequest(
+				getPaymentRequest( {
+					total: billing.cartTotal,
+					currencyCode: billing.currency.code.toLowerCase(),
+					countryCode: shippingData.shippingAddress.country,
+					shippingRequired: shippingData.needsShipping,
+					cartTotalItems: billing.cartTotalItems,
+					stripe,
+				} )
+			);
+		}
+		// otherwise we just update it (but only if payment processing hasn't
+		// already started)
+		if (
+			! paymentStatus.currentStatus.isPristine &&
+			currentPaymentRequest.current
+		) {
+			updatePaymentRequest( {
+				// @ts-ignore
+				paymentRequest: currentPaymentRequest.current,
+				total: billing.cartTotal,
+				currencyCode: billing.currency.code.toLowerCase(),
+				cartTotalItems: billing.cartTotalItems,
+			} );
+		}
+	}, [
+		billing.cartTotal,
+		billing.currency.code,
+		shippingData.shippingAddress.country,
+		shippingData.needsShipping,
+		billing.cartTotalItems,
+		paymentStatus.currentStatus.isPristine,
+		stripe,
+	] );
+
+	// whenever paymentRequest changes, then we need to update whether
+	// payment can be made
+	useEffect( () => {
+		if ( paymentRequest ) {
+			canDoApplePay( paymentRequest ).then( ( result ) => {
+				setCanMakePayment( result );
+			} );
+		}
+	}, [ paymentRequest ] );
+
+	// kick off payment processing
+	const onButtonClick = () => {
+		setActivePaymentMethod( PAYMENT_METHOD_NAME );
+		currentPaymentStatus.current.setPaymentStatus().processing();
+	};
+
+	const abortPayment = ( paymentMethod, message ) => {
```

Comment:
Not needed for this PR but we definitely should start adding docblocks to new methods.

Reply:
Agreed.